### PR TITLE
Add Whale's Split Keyboard(ver.3)

### PR DESCRIPTION
**Pull request by ncwhale**
Add my split keyboard firmware to QMK.

## Description

Add keyboard whale/sk/v3 codes.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* NONE

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/whale/sk/v3/info.json [hunk 1]
```diff
@@ -0,0 +1,970 @@
+{
```
Comment (zvecr) [inline keyboards/whale/sk/v3/info.json:1]: This file doesnt contain valid content
1. extra layout macros that do not exist in v3.h header
2. DEFAULT should be LAYOUT_all
3. DEFAULT thumb cluster order is invalid (tested with "fast input" mode on configurator)

File: keyboards/whale/sk/v3/rules.mk [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
```
[the remark this answers is not in the document]
Comment (ncwhale) [reply keyboards/whale/sk/v3/rules.mk:33]: I want to input emoji(unicode chars) from keyboard directly, how to enable this function when this flag set to no?

File: keyboards/whale/sk/v3/info.json (hunk 1 above)
Comment (ncwhale) [reply keyboards/whale/sk/v3/info.json:1]: I cannot find a way to validate this .json file, any suggestion?

File: keyboards/whale/sk/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Whale Mo
```
[the remark this answers is not in the document]
Comment (ncwhale) [reply keyboards/whale/sk/keymaps/default/config.h]: Done.

File: keyboards/whale/sk/v3/info.json (hunk 1 above)
Comment (ncwhale) [reply keyboards/whale/sk/v3/info.json:1]: I have both v3.h & info.json updated.

**Comment by ncwhale** [comment]
@zvecr THX for review my buggy codes.

But I still have one question: How to verify info.json file? I cannot found any document about verify it.

I added all the layout macros defined in v3.h file and it runs OK on my keyboard right now.

File: keyboards/whale/sk/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Splite Keyboard
```
Comment (fauxpark) [inline keyboards/whale/sk/README.md:1]: Please rename this file to lowercase.

File: keyboards/whale/sk/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    [_EMOJI] = LAYOUT_all(
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                                         XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                                                              XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX,
+                                                     XXXXXXX, XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, XXXXXXX,
+                                                              XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX
+    )
```
Comment (fauxpark) [inline keyboards/whale/sk/keymaps/default/keymap.c:66]:
```suggestion
```

File: keyboards/whale/sk/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN,
+    _MULTIMEDIA,
+    _EMOJI
```
Comment (fauxpark) [inline keyboards/whale/sk/keymaps/default/keymap.c:23]:
```suggestion
```

File: keyboards/whale/sk/v3/config.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER 0x0003
+#define MANUFACTURER Whale Mo
+#define PRODUCT Splite Keyboard v3
+#define DESCRIPTION Hackable split keyboard/QMK
```
Comment (fauxpark) [inline keyboards/whale/sk/v3/config.h:26]:
```suggestion
```

File: keyboards/whale/sk/v3/config.h [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS \
+    { B1, B2, B3, B4, B5, B6 }
+#define MATRIX_COL_PINS \
+    { D4, C6, D7, E6, F4, F5, F6 }
```
Comment (fauxpark) [inline keyboards/whale/sk/v3/config.h:46]:
```suggestion
#define MATRIX_ROW_PINS { B1, B2, B3, B4, B5, B6 }
#define MATRIX_COL_PINS { D4, C6, D7, E6, F4, F5, F6 }
```

File: keyboards/whale/sk/v3/v3.h [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    {L50, L51, L52, L53, L54, L55, L56, }, \
+    {R00, R01, R02, R03, R04, R05, R06, }, \
+    {R10, R11, R12, R13, R14, R15, R16, }, \
+    {R20, R21, R22, R23, R24, R25, R26, }, \
+    {R30, R31, R32, R33, R34, R35, R36, }, \
+    {R40, R41, R42, R43, R44, R45, R46, }, \
+    {R50, R51, R52, R53, R54, R55, R56 } \
+}
+
+#define LEFT_HALF_ONLY(\
```
Comment (noroadsleft) [inline keyboards/whale/sk/v3/v3.h:31]: QMK convention is that layout macros are prefixed `LAYOUT` in uppercase, with a lowercase description.

```suggestion
#define LAYOUT_left_half_only(\
```

File: keyboards/whale/sk/v3/v3.h [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+) { \
+    {L00, L01, L02, L03, L04, L05, L06, }, \
+    {L10, L11, L12, L13, L14, L15, L16, }, \
+    {L20, L21, L22, L23, L24, L25, L26, }, \
+    {L30, L31, L32, L33, L34, L35, L36, }, \
+    {L40, L41, L42, L43, L44, L45, L46, }, \
+    {L50, L51, L52, L53, L54, L55, L56 } \
+}
+
+#define RIGHT_HALF_ONLY( \
```
Comment (noroadsleft) [inline keyboards/whale/sk/v3/v3.h:49]:
```suggestion
#define LAYOUT_right_half_only( \
```

File: keyboards/whale/sk/v3/info.json [hunk 11]
```diff
@@ -0,0 +1,970 @@
...
+                    "label": "F12",
+                    "x": 11.5,
+                    "y": 6,
+                    "r": -30,
+                    "rx": 13,
+                    "ry": 3.5
+                }
+            ]
+        },
+        "LEFT_HALF_ONLY": {
```
Comment (noroadsleft) [inline keyboards/whale/sk/v3/info.json:487]:
```suggestion
        "LAYOUT_left_half_only": {
```

File: keyboards/whale/sk/v3/info.json [hunk 12]
```diff
@@ -0,0 +1,970 @@
...
+                    "label": "F6",
+                    "x": 8.5,
+                    "y": 6,
+                    "r": 30,
+                    "rx": 7,
+                    "ry": 3.5
+                }
+            ]
+        },
+        "RIGHT_HALF_ONLY": {
```
Comment (noroadsleft) [inline keyboards/whale/sk/v3/info.json:728]:
```suggestion
        "LAYOUT_right_half_only": {
```

File: keyboards/whale/sk/README.md (hunk 4 above)
Comment (ncwhale) [reply keyboards/whale/sk/README.md]: DONE.

File: keyboards/whale/sk/readme.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# Splite Keyboard
+
+![Split Keyboard Default Layout](https://github.com/ncwhale/Keyboard_Design/raw/master/Splite-Keyboard-v3/Keyboar-layout.png)
```
Comment (noroadsleft) [inline keyboards/whale/sk/readme.md:3]:
```suggestion
![Split Keyboard Default Layout](https://github.com/ncwhale/Keyboard_Design/raw/master/Splite-Keyboard-v3/Keyboard-layout.png)
```

**Comment by ncwhale** [comment]
@zvecr @fauxpark Hey guys, I thought most errors are fixed here, please review again?

**Pull request merged**